component/http/middleware/middleware.go

```diff
@@ -122,6 +124,21 @@ func NewRecovery() Func {
 	}
 }
 
+// NewAppNameVersion returns to all responses a header which contain the app name and version.
+func NewAppNameVersion(name, version string) Func {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if version != "" {
+				w.Header().Set(appVersionHeader, version)
+			}
+			if name != "" {
+				w.Header().Set(appNameHeader, name)
+			}
+			next.ServeHTTP(w, r)
+		})
+	}
```

Comment:
The proposed solutoin would breack the middleware signature i am afraid if we return an error.
It would make the usage a two step approach. 
First create the middleware then check for error and use it in the route vs just add it to the route.
If you check out all other middleware they all are not doing and defensive programming.

Reply:
> The proposed solutoin would breack the middleware signature i am afraid if we return an error.

It actually won't break the signature since the middleware interface is maintained in the 
`return func(next http.Handler) http.Handler {..}` statement.
The caller will need to check for the error before enabling/including the middleware in the pipeline

> If you check out all other middleware they all are not doing and defensive programming.

This is related to my second point in the above comment. If I'm not missing something, they don't do any error checking. For example, `Compression` does not validate the deflate level,  the `RequestObserver` does not validate the method and the path etc. I think that's ok btw, as long as we are consistent on how we approach this with all of our middlewares.

Reply:
Could it make sense to discuss this separately? Open a ticket to discuss this separately?
If we do error handling we need to change every middleware we have built ,which will be a breaking change since they are exported.

Reply:
Yes, as discussed we will be working on this in a separate issue to manage all middleware errors